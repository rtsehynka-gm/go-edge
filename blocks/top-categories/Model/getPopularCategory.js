import { fetchQuery } from '../../../scripts/commerce-backend.js';
import { GET_POPULAR_CATEGORY } from '../Queries/getPopularCategory.js';

async function getPopularCategory() {

  const { data } = await fetchQuery(GET_POPULAR_CATEGORY, {});
  console.log(data);
  return {
    items: data && data.categories && data.categories.items[0] && data.categories.items[0].popular_category,
  };
}

export {
  getPopularCategory,
};
