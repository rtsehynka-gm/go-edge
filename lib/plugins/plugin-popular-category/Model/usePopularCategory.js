import { useQuery } from "../../../../scripts/commerce-backend.js";
import { GET_POPULAR_CATEGORY } from '../Queries/getPopularCategory.gql';

/**
 *
 * @returns {{items: *}}
 */
export const usePopularCategory = async () => {
  const { data } = await useQuery(GET_POPULAR_CATEGORY);
  return { items: data?.categories?.items[0]?.popular_category };
};
