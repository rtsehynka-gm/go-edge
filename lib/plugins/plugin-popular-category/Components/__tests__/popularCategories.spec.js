import React from 'react';
import { createTestInstance } from '@magento/peregrine';
import PopularCategories from '../popularCategories';
import { usePopularCategory } from '../../Model/usePopularCategory';

jest.mock('@magento/venia-ui/lib/classify');
jest.mock('../../Model/usePopularCategory', () => ({
    usePopularCategory: jest.fn()
}));
jest.mock('react-intl', () => ({
    FormattedMessage: props => (
        <div componentName="Formatted Message Component" {...props} />
    )
}));
jest.mock('@magento/venia-ui/lib/components/Image', () => props => (
    <div componentName="Image" {...props} />
));
jest.mock('react-router-dom', () => ({
    Link: ({ children }) => children
}));
const talonProps = {
    items: [
        {
            name: 'test Category',
            url_key: '/testCategory.html',
            url_path:'/testCategory.html',
            popular_category_image: 'imageTest.png',
            show_popular_category: true
        },
        {
            name: 'test Category2',
            url_key: '/testCategory2.html',
            url_path:'/testCategory2.html',
            popular_category_image: 'imageTest2.png',
            show_popular_category: false
        },
        {
            name: 'test Category3',
            url_key: '/testCategory3.html',
            url_path:'/testCategory3.html',
            popular_category_image: 'imageTest3.png',
            show_popular_category: true
        }
    ]
};

test('renders without items', () => {
    const props = {
        data: {}
    }
    usePopularCategory.mockReturnValueOnce({items: null});
    const component = createTestInstance(<PopularCategories {...props} />);
    expect(component.toJSON()).toMatchSnapshot();
});

test('renders with items', () => {
    const props = {
        data: {}
    }
    usePopularCategory.mockReturnValueOnce(talonProps);
    const component = createTestInstance(<PopularCategories {...props} />);
    expect(component.toJSON()).toMatchSnapshot();
});

