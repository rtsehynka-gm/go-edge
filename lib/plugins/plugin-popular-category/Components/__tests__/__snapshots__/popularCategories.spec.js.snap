// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with items 1`] = `
<div
  className="root"
>
  <h2
    className="title"
  >
    <div
      componentName="Formatted Message Component"
      defaultMessage="Popular Categories"
      id="popularCategories.title"
    />
  </h2>
  <div
    className="items"
  >
    <div
      className="item"
    >
      <span
        className="linkTitle"
      >
        <div
          componentName="Formatted Message Component"
          defaultMessage="view more"
          id="popularCategories.link"
        />
      </span>
      <div
        alt="test Category"
        classes={
          Object {
            "root": "container",
          }
        }
        componentName="Image"
        height={200}
        src="imageTest.png"
        width={300}
      />
      <p
        className="itemName"
      >
        test Category
      </p>
    </div>
    <div
      className="item"
    >
      <span
        className="linkTitle"
      >
        <div
          componentName="Formatted Message Component"
          defaultMessage="view more"
          id="popularCategories.link"
        />
      </span>
      <div
        alt="test Category3"
        classes={
          Object {
            "root": "container",
          }
        }
        componentName="Image"
        height={200}
        src="imageTest3.png"
        width={300}
      />
      <p
        className="itemName"
      >
        test Category3
      </p>
    </div>
  </div>
</div>
`;

exports[`renders without items 1`] = `null`;
